Sources/Sentry/SentryScreenshotIntegration.m

```diff
@@ -16,15 +16,26 @@
 saveScreenShot(const char *path)
 {
     NSString *reportPath = [NSString stringWithUTF8String:path];
-    if (![NSFileManager.defaultManager fileExistsAtPath:reportPath isDirectory:nil]) {
-        NSError *error = nil;
+    NSError *error = nil;
+
+    if (![NSFileManager.defaultManager fileExistsAtPath:reportPath]) {
         [NSFileManager.defaultManager createDirectoryAtPath:reportPath
                                 withIntermediateDirectories:YES
                                                  attributes:nil
                                                       error:&error];
         if (error != nil)
             return;
     }
+
+    // We first delete any screenshot that could be from an old crash report
+    NSArray *oldFiles = [NSFileManager.defaultManager contentsOfDirectoryAtPath:reportPath
```

Comment:
`h`: Can't we design this in a way, that we can have multiple crash reports and multiple screenshots?

Reply:
That's exactly what we are doing. SentryCrash creates a unique name for each report (at least it should) and we create a screenshot directory for each report as well. 

This `delete` feature is just in case SentryCrash, for some reason, reuses a report name.

Reply:
Yes. Im aware of that. Just added the delete feature for safety. Do you think we should remove it?

Reply:
I dont know the inner works of _Atomic, but what I found out if we do this:

```objc
    uint32_t lowerBaseID = (uint32_t)(baseID & 0xffffffff);
    g_nextUniqueIDLow = lowerBaseID;
```

It works, otherwise we always get the same value.